lib/src/rules/missing_conditional_import.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Missing conditional import.';
+
+const _details = r'''
+
+**DON'T** reference files that do not exist in conditional imports.
+
+Code may fail at runtime if the condition evaluates such that the missing file
+needs to be imported.
+
+**BAD:**
+```dart
+import 'file_that_does_not_exist.dart';
+```
+
+**GOOD:**
+```dart
+import 'file_that_exists.dart';
+```
+
+''';
+
+class MissingConditionalImport extends LintRule {
+  MissingConditionalImport()
+      : super(
+            name: 'missing_conditional_import',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    var visitor = _Visitor(this);
+    registry.addCompilationUnit(this, visitor);
+  }
+}
+
+class _Visitor extends RecursiveAstVisitor<void> {
+  static const LintCode code = LintCode('missing_conditional_import',
+      "The target of the conditional URI '{0}' doesn't exist.",
+      correctionMessage: 'Try creating the file referenced by the URI, or '
+          'Try using a URI for a file that does exist.');
+
+  final LintRule rule;
+
+  _Visitor(this.rule);
+
+  @override
+  void visitConfiguration(Configuration configuration) {
+    String? uriContent = configuration.uri.stringValue;
+    if (uriContent != null) {
+      Source? source = configuration.uriSource;
+      if (!(source?.exists() ?? false)) {
```

Comment:
As you noted, this will probably work most of the time, but it would be better if we could ask the resource provider. Unfortunately there doesn't appear to be any way to access it currently.

My personal opinion is that we need to update `LinterContext` to provide access to the resource provider and then pass that in when constructing the visitor. Unfortunately, that's a multi-step process, and I'm not sure we want to delay merging this rule while that happens.

Reply:
I'm happy to go with your decision here. If you think it's worth doing (and I know enough to do it), I'm happy to. Although I'm not sure if having to support `package:` URIs complicates using the resource provider. This `exists()` call seemed to handle that (and `dart:` URIs) for me which I thought might have complicated things :-)

Reply:
I'm fine with leaving it as-is. You might consider adding a comment for future reference, because the fact that it isn't checking for overlays might not occur to a future reader.

Reply:
Done!

Would this lint be invalidated/updated correctly if the user saved a new file with the name that was used here? Would the server know if a file has `import 'file_a.dart'` but that file does not exist, that creating a file at that location may require this file to be re-analyzed/linted?

Reply:
I don't know whether it _is_ the case, but it ought to be re-analyzed when a new file is added, and we run lints as part of analyzing a file.